composer/utils/object_store/oci_object_store.py

```diff
@@ -130,24 +144,48 @@ def download_object(
         dirname = os.path.dirname(filename)
         if dirname:
             os.makedirs(dirname, exist_ok=True)
-        tmp_path = str(filename) + f'.{uuid.uuid4()}.tmp'
 
-        try:
-            response = self.client.get_object(
-                namespace_name=self.namespace,
-                bucket_name=self.bucket,
-                object_name=object_name,
-            )
-        except Exception as e:
-            _reraise_oci_errors(self.get_uri(object_name), e)
-
-        with open(tmp_path, 'wb') as f:
-            f.write(response.data.content)
-
-        if overwrite:
-            os.replace(tmp_path, filename)
-        else:
-            os.rename(tmp_path, filename)
+        # Get the size of the object
+        head_object_response = self.client.head_object(self.namespace, self.bucket, object_name)
+        object_size = head_object_response.headers['content-length']
+        # Calculate the part sizes
+        base_part_size = int(object_size) // num_parts
+        remainder = int(object_size) % num_parts
```

Comment:
these two lines can be simplified using `divmod` I believe

Reply:
Thanks! Changed it to use `divmod`.